modelconverter/cli/types.py

```diff
@@ -1,38 +1,13 @@
 from enum import Enum
-from pathlib import Path
 from typing import List, Optional
 
+import click
 import typer
 from typing_extensions import Annotated
 
 from modelconverter.utils.types import Target
 
 
-class Task(str, Enum):
```

Comment:
What is the reason for dropping these enums? Because they are shown by click choices? From code quiality standpoint I feel like enums do help keep everything more organized.

Reply:
tldr; they're only used as a way to limit CLI some arguments becuase `typer` doesn't support the `Literal` type. 

---

`typer` that we're using to build the CLI doesn't support the `Literal` type in function annotations. The only way how to get a nice support for an argument with a set of restricted values is to create an enum, so the enums are only used for this. This makes using the functions outside of the CLI a bit cumbersome (which started to be an issue with the online conversion). 
 
There's an open issue and even a PR to add the `Literal` support for several years but it seems somehow ignored. One of the comments there showed how to achieve a bit similar behavior to `Literal` by leveraging the underlying `click` library that `typer` is built on. 

Alternative way would be to have two sets of functions. One that would be more simple to use programmatically (with `Literal`s) and an almost identical set of signatures with enums for the CLI (implementation-wise thye'd be just wrappers around the first set). From the user perspective this would probably be the cleanest, but it would be a bit ugly codebase-wise imo.

Reply:
Ok I see I see.. yeah in that case let's leave it as is